api/src/identity-access-management/domain/errors.js

```diff
@@ -89,9 +89,17 @@ class UserShouldChangePasswordError extends DomainError {
   }
 }
 
+class InvalidLtiPlatformRegistrationError extends DomainError {
+  constructor(message = 'LTI platform registration is invalid', { cause } = {}) {
+    super(message, 'INVALID_LTI_PLATFORM_REGISTRATION');
+    this.cause = cause;
+  }
+}
```

Comment:
Pour la définition suivre le doc spécifique https://github.com/1024pix/pix/blob/dev/docs/errors.md et l'ADR https://github.com/1024pix/pix/blob/dev/docs/adr/0044-gestion-erreurs-i18n-reference.md, ce qui donne la proposition ci-dessous.

Les principes sont que on ne veut pas qu'il soit pensé que le message est modifiable, et il a été standardisé le fait de passer des infos spécifiques dans une propriété `meta`. C'est un peu moins pratique que d'avoir directement la propriété `cause` mais c'est générique.

```suggestion
class InvalidLtiPlatformRegistrationError extends DomainError {
  constructor({ meta } = {}) {
    super('LTI platform registration is invalid', 'INVALID_LTI_PLATFORM_REGISTRATION');
    if (meta?.cause) {
        this.meta = { cause: meta.cause };
    }
  }
}
```

Reply:
[Error.cause](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause) fait partie de JavaScript, ça permet notamment d'avoir des stacktraces complètes lorsqu'une erreur est catchée et re-throwée sous une autre forme.
Du coup ça paraît contreproductif pour le debug d'abandonner cette feature dans nos erreurs.
Plus généralement, avoir des signatures de constructeur d'erreur qui s'éloignent de `new Error(message, options)`, alors que celui-ci permet déjà de faire ce qu'on veut avec `options`, paraît contreproductif pour que les gens puissent s'approprier le code facilement.

Le [constructeur de `DomainError`](https://github.com/1024pix/pix/blob/4540608e2c86f704b178b8bb044605349af35192/api/src/shared/domain/errors.js#L3) est assez surprenant de ce point de vue, il empêche d'utiliser l'argument `options` de sa classe parente, et le remplace par des arguments customs.
Mon avis personnel, c'est qu'on devrait plutôt revoir ça.

Pour revenir à notre code, notre usecase change le message de l'erreur afin de donner plus d'informations à l'utilisateur, est-ce que c'est une mauvaise pratique ?
Ou bien faut-il créer une classe d'erreur pour chaque message ?

Reply:
Bien reçu. Il faudrait alors vraiment faire évoluer https://github.com/1024pix/pix/blob/dev/docs/errors.md ! Mais à discuter dans un autre contexte !

Et effectivement je n'avais pas vérifier que `InvalidLtiPlatformRegistrationError` est utilisée à plusieurs endroits avec des messages différents. Il est donc légitime de pouvoir changer le message dans le constructeur. Je marque donc cette discussion comme résolue. Merci !